tests/unit_tests/test_unit.py

```
@@ -577,6 +577,51 @@ def test_calc_match_score():
     )
     assert np.allclose(np.array([1.0]), aa_scores[4])
 
+    # Reverse direction.
+    flipped_batch_all_aa_scores = torch.flip(batch_all_aa_scores, dims=[1])
+    flipped_true_aas = torch.flip(true_aas, dims=[1])
+    pep_scores_reversed, aa_scores_reversed = _calc_match_score(
+        flipped_batch_all_aa_scores, flipped_true_aas
+    )
+
+    assert all([pytest.approx(0.0) == x for x in pep_scores_reversed])
```

Ruff (0.15.6)

[warning] Unnecessary list comprehension

Remove unnecessary comprehension

(C419)